dotnet/xwalk_dotnet_bridge.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#ifndef XWALK_DOTNET_XWALK_DOTNET_BRIDGE_H_
+#define XWALK_DOTNET_XWALK_DOTNET_BRIDGE_H_
+
+#include <string>
+
+#include "xwalk/extensions/public/XW_Extension.h"
+#include "xwalk/extensions/public/XW_Extension_EntryPoints.h"
+#include "xwalk/extensions/public/XW_Extension_Permissions.h"
+#include "xwalk/extensions/public/XW_Extension_Runtime.h"
+#include "xwalk/extensions/public/XW_Extension_SyncMessage.h"
+
+namespace xwalk {
+namespace extensions {
+
+class XWalkDotNetBridge {
+ public:
+  XWalkDotNetBridge();
+  ~XWalkDotNetBridge();
+
+  // XW_Extension callbacks.
+  static void OnShutdown(XW_Extension xw_extension);
+  static void OnInstanceCreated(XW_Instance xw_instance);
+  static void OnInstanceDestroyed(XW_Instance xw_instance);
+  static void HandleMessage(XW_Instance xw_instance, const char* msg);
+  static void HandleSyncMessage(XW_Instance xw_instance, const char* msg);
+
+  bool Initialize();
+  void* CreateInstance(XW_Instance native_instance);
+  void PostMessageToInstance(XW_Instance instance, const std::string& message);
+  void SetSyncReply(XW_Instance instance, const std::string& message);
+
+ private:
+  std::string library_path_;
+  bool initialized_;
+  void* extension_dotnet_;
+  void* extension_assembly_;
```

Comment:
how about struct PrivateData { gcroot<Object^> extension_object; ... extension_assembly; } // in .cc file
and here in class declaration:
PrivateData* p_data_;